jujugui/static/gui/src/app/jujulib/charmstore.js

```diff
@@ -83,6 +83,10 @@ var module = module;
     _transformQueryResults: function(callback, error, data) {
       if (error !== null) {
         callback(error, data);
+      } else if (!data) {
+        // The only time there should be no data and no error is if the
```

Comment:
As this comment explains, when retrieving an entity from the charm store we can always assume that the response contains data or it is a 404. An empty response is never a legitimate response from that API endpoint, therefore maybe we should turn that case into an error at the API client level. Take it or leave it.